### PR TITLE
Fix invalid progress calculation

### PR DESCRIPTION
**Pull request by ianthetechie**
This is probably the root cause of #126.

### CODE REVIEW

File: common/ferrostar/src/algorithms.rs [hunk 1]
```diff
@@ -273,28 +273,33 @@ fn distance_to_end_of_step(snapped_location: &Point, current_step_linestring: &L
...
+///
+/// NOTE: `remaining_steps` includes the current step.
 pub fn calculate_trip_progress(
     snapped_location: &Point,
-    current_step: &RouteStep,
     current_step_linestring: &LineString,
     remaining_steps: &[RouteStep],
 ) -> TripProgress {
-    if remaining_steps.is_empty() {
+    let Some(current_step) = remaining_steps.first() else {
```
Comment (ianthetechie) [inline common/ferrostar/src/algorithms.rs:284]: Passing current_step explicitly here opened the door for potentially very confusing errors caused by this ambiguity. I've just made it impossible to make that mistake now.

File: common/ferrostar/Cargo.toml [hunk 2]
```diff
@@ -43,7 +43,4 @@ insta = { version = "1.33.0", features = ["yaml"] }
 rstest = "0.21.0"
 
 [lib]
-crate-type = ["cdylib", "staticlib", "lib"]
-
-[profile.release]
-lto = "thin"
\ No newline at end of file
+crate-type = ["cdylib", "staticlib", "lib"]
```
Comment (ianthetechie) [inline common/ferrostar/Cargo.toml:46]: Linter catch: this didn't have any effect in a workspace sub-project ;)

File: common/build-ios.sh [hunk 3]
```diff
@@ -63,11 +63,12 @@ build_xcframework() {
...
 
-cargo build --lib --release --target x86_64-apple-ios
-cargo build --lib --release --target aarch64-apple-ios-sim
-cargo build --lib --release --target aarch64-apple-ios
-
 basename=ferrostar
+
+cargo build -p $basename --lib --release --target x86_64-apple-ios
+cargo build -p $basename --lib --release --target aarch64-apple-ios-sim
+cargo build -p $basename --lib --release --target aarch64-apple-ios
```
Comment (ianthetechie) [inline common/build-ios.sh:70]: Minor improvement: we don't need to build `uniffi-bindgen` for any other architectures.

File: common/ferrostar/src/algorithms.rs [hunk 4]
```diff
@@ -262,39 +263,43 @@ fn distance_along(point: &Point, linestring: &LineString) -> Option<f64> {
...
 
     // This could be improved with live traffic data along the route.
     // TODO: Figure out the best way to enable this use case
-    let pct_remaining_current_step =
-        distance_to_next_maneuver / current_step_linestring.haversine_length();
+    let pct_remaining_current_step = if current_step.distance > 0f64 {
+        distance_to_next_maneuver / current_step.distance
+    } else {
+        0f64
+    };
```
Comment (ianthetechie) [inline common/ferrostar/src/algorithms.rs:302]: This was the core issue. If the snapped point is exactly at the end of the step, this results in division by zero.

File: common/ferrostar/src/algorithms.rs [hunk 5]
```diff
@@ -476,6 +481,37 @@ proptest! {
...
+        prop_assert_eq!(progress.duration_remaining, 0f64);
+    }
+
+    #[test]
+    fn test_end_of_trip_progress_valhalla_arrival(
+        x1: f64, y1: f64,
+    ) {
+        // This may look wrong, but it's actually how Valhalla (and presumably others)
+        // represent a point geometry for the arrival step.
+        let current_route_step = gen_dummy_route_step(x1, y1, x1, y1);
```
Comment (ianthetechie) [inline common/ferrostar/src/algorithms.rs:506]: And this is why I noticed the issue; the last maneuver's geometry is the same point twice, so its length is zero; thus, the division by zero above, which ended up crashing one of the iOS formatters in the arrival view with a `NaN`.

**Review by Archdoog — APPROVED** [review]

**Pull request merged**